### PR TITLE
UI improvements

### LINKED ISSUE #88: re-organise mobile view

### PR DESCRIPTION
**Pull request by AdamFipke**
# Description

Basically, a boat load of mostly smallish tweaks and improvements to the UI on the queue page (especially mobile) and on the today page. See screenshots below. Below each screenshot is some text to show what has changed. I wouldn't spend too much time bothering to review the code thoroughly for this one since it's unlikely anything is broken catastrophically (and would be easy for me to fix later anyway) and I don't think my code made the current frontend spaghetti much worse than it already was lol (like I actually made sure to add comments, meanwhile I don't think I saw a single one and some things were really hard to follow lol). 

This PR mostly just involves adjusting/adding styles to various elements, with some additional logic to add styles based on variables already in the class (i.e. there's no additional hook or server calls)

**main goal:** increase usability, decrease cognitive load for main use cases

Closes #88 

# Screenshots

### Desktop
image
- adjusted margins around queue a bit
- added an icon to join queue
- Increased boldness of queue name header and decreased boldness of staff header
- italicized the "queue is up to date" widget 

image
- added highlight to your questions in the queue (to make it easily glancible where you are)

image
- Shrunk avatar of current checked in staff to be less colossal
- Queues of current checked in staff now have the following (in order to help make it obvious for students to find what queue they probably want to go into):
	- ~~buttons become primary blue~~
	- animated rainbow border
- moved the queue tags (i.e. "professor queue" or "not accepting questions") to be closer to the queue title rather than close to the button (that way it's intuitively more of a description of the queue rather than a description of the button, plus it cleans up the area around the button to be less distracting)
- Removed some not-helpful text in an effort to reduce visual clutter: 
	- Queues with no notes provided no longer has "*no notes provided" text
	- Async question centre no longer has "staff checked in" text (I'm not really sure why it was there to begin with?)
- renamed "Async Question center" -> "Async Question Centre" (because Canada, eh)
- ~~Added little ">" arrows to the open queue buttons, might change it to a different icon later idk~~
- **Made the queue cards clickable**
	- **removed the 'Open Queue' buttons**
	- **has nice hover effects**
		- **drop shadow, card header lightens, arrow turns green (not sure about how to animate the arrow yet, too much animation is distracting but too little doesn't give additional help to convey that it's clickable)**
	- **gave mobile view a bit more dropdown shadow to make it appear more tappable** (should work well with the rainbow border)
- **Made async queue centre slightly purple to help differentiate it from the regular queues**
- **Moved async queue's "Ask your questions anytime!" text into the card content"
- **Made checked-in staff appear to the right of the 'Current Checked In Staff' text instead of below**

### Mobile
image
- Made use of empty page header:
	- Added course name as the main header
	- Current Queue name as the subheader (obviously not visible if you're not in a queue)
- Moved "Queue is up to date" text to underneath the "Queue" header, italicized it, and reduced its size
- Moved "Join Queue" button to be right beside the "Queue" header and made it less wide
- Adjusted margins all around so there's less gap between staff section and queue section
- Fixed alignment of the hamburger icon (top right nav button)
- hid footer on mobile
- moved the chatbot button to be not hugging the left side
- Reduced the padding in the staff queue cards so they take up just a little less space
- Reduced the padding of the student questions just a little

All of the changes above now result in the Queue starting in the top third of the screen rather the bottom third.

image
- Made adjustments to the "you are X in queue" banner
	- It now appears under the "Queue" header
	- Reduced the massive vertical margins around it 
	- Fixed the banner buttons going vertical
- Also, reduced the size and margins around the queue notes, which now also appear at the top (wasn't really sure where to put it exactly, though there's a good chance that the queue notes will be unset anyways. If you have any better ideas on where it could be put feel free to let me know)

image
- Centered "Queue" text
- changed navbar title from "Course" -> course logo + course name
image
- Made avatar a bit larger
- Aligned avatar with other nav elements
- Fixed the dropdown appearing on the right instead of under the avatar on the left

image
- Fixed the positioning of the dropdown to be on the right instead of left
- Tapping on a different queue will now properly close the whole nav drawer (before it wouldn't, meaning you would need to manually close the drawer after switching queues)

### Desktop
image
- All the stuff mentioned on the student side plus:
	- Edit notes button changed to an 'edit' icon button (reduce reading, easier cognitive load)
	- "*no notes provided*" text made a little more light gray
	- Icons added to checkin/checkout button (same on queue page)

image
- Renamed "Add Students" button to "Add Students to Queue" to make it more obvious
- Added icons to the "Edit Queue Details" and "Add Students to Queue" buttons
- Removed the "Help Next" button since now it provides very little utility for how much visual noise it adds (it made sense with the old system where it require multiple clicks to help the next person but it's kinda redundant now)
- Check out button moved to top (probably more frequently used use-case)

image
- Adjusted and standardised margins between form items
- Adjusted and added margins between a lot items, including around the "Current Question Types" stuff and the zoom link stuff
- Made form item titles bold
- Made the current zoom link actually clickable
- Put colour picker, question type name, and add button all on the same line (question types don't get very long anyways)

image
- Colour picker now appears below like so, rather than pushing the add button down

image
- removed redundant "Search for students to be added" text

### Mobile

image
In addition to what was listed on the student mobile changes:
- No longer a column of 6 buttons!
	- Check out, Edit Queue, and Add Students button are now all on the same row
	- As stated previously, Help Next was removed
	- Clear Queue and Delete Queue have been moved into the Edit Queue modal (but only for mobile), which not only gives more screen space but also makes it less easy to accidentally perform those actions on mobile
- "Edit Queue Details" gets shortened to "Edit Queue" on mobile
- "Add Students to Queue" gets shortened to "+ Students" on mobile
- The 3 buttons on students' question cards are now spread apart to make it harder to 'fat-thumb' the wrong button
- Other margin adjustments for extra screen space
- Changes to a lot of elements to use 'rem' or 'em' units rather 'px' to make it more dynamic for smaller phones

With the combined changes above, the "Queue" now starts above halfway at the screen rather than at the very bottom (or not visible at all, see below with an iPhone 12 Pro display, previous screenshots were with a Pixel 7)
image

image
- (showcasing the moved "clear" and "delete queue" buttons)

+ probably some other small changes I forgot

#### ***  NEW  ***

image
- Removed "Hotkeys" from dropdown
- Fixed its margins (antd gave it like 50% more margins on the left side)

- Edited the today page (specifically the queue cards) a lot, check out the new screenshots and changes (new stuff will be bolded) above. -> NOTE TO SELF: still need to do this

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This requires a run of `yarn install`

# How Has This Been Tested?

Manual testing. Checking to see if the changes broke any related components. 

Since this is all UI changes, this won't break anything in the backend and if there are any issues that you guys find, it's easy for me to fix.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code where needed 
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/app/components/Today/QueueCard.tsx [hunk 1]
```diff
@@ -67,96 +72,118 @@ const HeaderText = styled.div`
...
 
   const handleUpdate = () => {
-    setEditingNotes(false);
-    updateQueueNotes(queue, updatedNotes);
-  };
+    setEditingNotes(false)
+    updateQueueNotes(queue, updatedNotes)
+  }
   return (
     <PaddedCard
```
Comment (Ferdinand737) [inline packages/app/components/Today/QueueCard.tsx:154]: What if we made the whole card clickable? Just an idea.

**Review by Ferdinand737 — APPROVED** [review]
Overall really nice job!

Huge improvments to usability and style

File: packages/app/components/Today/QueueCard.tsx (hunk 1 above)
Comment (AdamFipke) [reply packages/app/components/Today/QueueCard.tsx:154]: Ooh good idea! That would certainly make it easier to navigate to the queue since they wouldn't need to find the button. Plus removing the button would mean less visual clutter, which is always a good thing. I shall investigate this later to see if it would break anything
